### PR TITLE
dry-run: Update DynamicClient to pass Create/Update options

### PR DESCRIPTION
**Pull request by apelisse**
Change dynamic client to accept Create and Update options on Create/Update/Patch methods.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/garbagecollector/operations.go [hunk 1]
```diff
@@ -73,15 +73,15 @@ func (gc *GarbageCollector) updateObject(item objectReference, obj *unstructured
 	if err != nil {
 		return nil, err
 	}
-	return gc.dynamicClient.Resource(resource).Namespace(resourceDefaultNamespace(namespaced, item.Namespace)).Update(obj)
+	return gc.dynamicClient.Resource(resource).Namespace(resourceDefaultNamespace(namespaced, item.Namespace)).Update(obj, &metav1.UpdateOptions{})
```
Comment (deads2k) [inline pkg/controller/garbagecollector/operations.go]: nil doesn't express no opinion?

**Comment by deads2k** [comment]
Actually, reading the pattern in other clients, it appears that query params are non-pointers and body values are pointers.  When you create set it up for the generated clients, which way do you plan to go?

I'd like to see one of two changes:
 1. pass nil for a pointer value (most places do this for delete).
 2. be consistent with query params and pass structs.

I don't really care which.  lgtm otherwise

/approve

**Comment by apelisse** [comment]
OK David, removed the pointer everywhere, should be good now.

File: staging/src/k8s.io/client-go/dynamic/interface.go [hunk 2]
```diff
@@ -29,15 +29,15 @@ type Interface interface {
 }
 
 type ResourceInterface interface {
-	Create(obj *unstructured.Unstructured, subresources ...string) (*unstructured.Unstructured, error)
-	Update(obj *unstructured.Unstructured, subresources ...string) (*unstructured.Unstructured, error)
+	Create(obj *unstructured.Unstructured, options metav1.CreateOptions, subresources ...string) (*unstructured.Unstructured, error)
+	Update(obj *unstructured.Unstructured, options metav1.UpdateOptions, subresources ...string) (*unstructured.Unstructured, error)
 	UpdateStatus(obj *unstructured.Unstructured) (*unstructured.Unstructured, error)
```
Comment (deads2k) [inline staging/src/k8s.io/client-go/dynamic/interface.go]: Sorry, I missed this on my first pass.  This one should be able to pass updateoptions too.

Comment (apelisse) [reply staging/src/k8s.io/client-go/dynamic/interface.go]: Good catch. I can't remember why I thought we wouldn't do this one ...

**Pull request merged**